### PR TITLE
Python simpler distribtests

### PR DESCRIPTION
**Pull request by jtattermusch**
Currently running the artifacts - packages - distribtests flow needs to be done manually, it's a hassle to do it and it takes many hours for the flow to finish (and after that interpreting the results is involves lots of clicking on links in the test results).

Since as of now the python wheels built on linux  are only tested by distribtests on linux, it is possible to run the entire python-specific artifact - package - distribtests flow for linux in a single kokoro job.

So this new job first builds python linux wheels (and source archives) and then tests them with distribtests in a single kokoro job.
Advantages:
- the test results are easy to interpret
- the job can run on PRs
- the test script basically shows how to run the test python wheels with distribtest locally on dev's workstation.

### CODE REVIEW

**Comment by jtattermusch** [comment]
@lidizheng  @gnossen  @drfloob  this is not fully ready to review, but the main idea is already there.

File: tools/internal_ci/linux/grpc_distribtests_python.cfg
Comment (drfloob) [inline tools/internal_ci/linux/grpc_distribtests_python.cfg:1]: Copyright 2021

File: tools/internal_ci/linux/grpc_distribtests_python.cfg
Comment (drfloob) [inline tools/internal_ci/linux/grpc_distribtests_python.cfg:19]: Do we really need 6 hours for this test?

File: tools/internal_ci/linux/grpc_distribtests_python.sh
Comment (drfloob) [inline tools/internal_ci/linux/grpc_distribtests_python.sh:29]: Will the other steps below fail fast if the artifacts fail to build? It might be worth exiting here on failure to speed things up.

File: tools/internal_ci/linux/grpc_distribtests_standalone.cfg
Comment (drfloob) [inline tools/internal_ci/linux/grpc_distribtests_standalone.cfg:19]: same question here regarding 6hrs

**Review by drfloob — COMMENTED** [review]
This is great, Jan, thanks for putting it together. This job is only used for manual runs, right? Unless I'm mistaken there, it might be worth leaving a comment to that effect in the CI config file.

File: tools/internal_ci/linux/grpc_distribtests_python.cfg
Comment (lidizheng) [reply tools/internal_ci/linux/grpc_distribtests_python.cfg:19]: Actually... is 6 hours enough? If we are compiling ARM wheels with QEMU in this job, the total time consumption will be at least 7 hours.

Here is a timeout Python artifact ARM build (>7 hours): https://fusion.corp.google.com/runanalysis/info/prod:grpc%2Fcore%2Fmaster%2Flinux%2Fgrpc_build_artifacts_extra/prod:grpc%2Fcore%2Fmaster%2Flinux%2Fgrpc_build_artifacts_extra/KOKORO/7702989d-368e-4f31-9078-2125869f70f3/1615801227889/build%20%237985/Targets

File: tools/run_tests/task_runner.py
Comment (lidizheng) [inline tools/run_tests/task_runner.py:80]: Maybe we need to use the `xml_report` somewhere?

https://github.com/grpc/grpc/blob/dd54a74fefb2065549b409862da701c7e68acd3c/tools/run_tests/task_runner.py#L117

File: tools/internal_ci/linux/grpc_distribtests_python.cfg
Comment (jtattermusch) [reply tools/internal_ci/linux/grpc_distribtests_python.cfg:19]: The last 2 runs I made (by "kidnapping" and existing PR test job) took between 2 and 2.5 hours. While that's not great, it's also not terrible and it's much faster than the full multilanguage artifacts - packages - distribtest flow.

I can reduce the timeout to 240mins to still allow some buffer.

On ARM7 build: we actually no longer run these, in one of my relatively recent PRs I changed the "linux_extra" so that they are crosscompiled (instead of building under an emulator), which is almost as fast as a regular build.

File: tools/internal_ci/linux/grpc_distribtests_standalone.cfg
Comment (jtattermusch) [reply tools/internal_ci/linux/grpc_distribtests_standalone.cfg:19]: This change is just a trick to "kidnap" an existing PR job to be able to get the test results. (It's the "DO NOT MERGE" commit in the commit history). I will remove this commit before merging.

File: tools/run_tests/task_runner.py
Comment (jtattermusch) [reply tools/run_tests/task_runner.py:80]: :-))) yes that's a good point. I was wondering why the reports don't show up in the way I wanted :-).
Good catch.

File: tools/internal_ci/linux/grpc_distribtests_python.sh
Comment (jtattermusch) [reply tools/internal_ci/linux/grpc_distribtests_python.sh:29]: I think we quite often have situations where only one of the artifact builds gets broken, but the others will build just fine. Running the distribtests in that case gives you a good picture which distribtests are impacted by that and which are not (and I think that's a useful signal). 
AFAIK the distribtests will fail quite fast (in like a minute is so) when the wheel required by a particular distribtest is missing.

File: tools/internal_ci/linux/grpc_distribtests_python.cfg
Comment (jtattermusch) [reply tools/internal_ci/linux/grpc_distribtests_python.cfg]: Adjusted to 240mins.

**Comment by jtattermusch** [comment]
This is an example of how the test results would look like (obtained by kidnapping another PR job temporarily):
https://source.cloud.google.com/results/invocations/d7628d85-6ecc-479e-bfa3-c9586ff4075d/targets

**Comment by jtattermusch** [comment]
I believe I've addressed all the comments and I also removed the "DO NOT MERGE" commit. This should now be ready for final review (and merge ;-) )

**Review by drfloob — APPROVED** [review]

**Pull request merged**
